src/dxFeedNativeAPI.cpp

```diff
@@ -3,4 +3,35 @@
 #include <dxFeedNativeAPI.h>
 #include <dxFeedNativeCppAPI.hpp>
 
-namespace dxfcpp {}
+#include <cstring>
+#include <utf8cpp/utf8.h>
+
+dxfc_error_code_t dxfc_system_set_property(const char *key, const char *value) {
+    return dxfcpp::System::setProperty(key, value) ? DXFC_EC_SUCCESS : DXFC_EC_ERROR;
+}
+
+dxfc_error_code_t dxfc_system_get_property(const char *key, char *buffer, size_t bufferSize) {
+    auto value = dxfcpp::System::getProperty(key);
+
+    if (buffer != nullptr && bufferSize > 0ULL) {
+        auto size = std::min(value.size(), bufferSize - 1);
+
+        buffer[size] = '\0';
+
+        if (size == 0ULL) {
+            return DXFC_EC_SUCCESS;
+        }
+
+        memcpy(buffer, value.c_str(), size);
+
+        auto begin = buffer;
+        auto end = buffer + size;
+        auto found = utf8::find_invalid(begin, end);
+
+        if (found != end) {
+            *found = '\0';
+        }
```

Comment:
Is this result supposed to be used somehow yet?

Reply:
Here we look for the position of an invalid UTF-8 character (because the string may not fit in the buffer) and truncate the string to that invalid character. Or I didn't understand what you mean =)

Reply:
Sorry, I didn't see "found" was an iterator)